### DISCUSSION TITLE
Donation drive to implement diarization [goal 100£]

### DISCUSSION THREAD

**Discussion by Purfview** [Purfview/whisper-standalone-win#281] [Ideas]
Donation drive to implement diarization [goal 100£]

Derived from this post: https://github.com/Purfview/whisper-standalone-win/discussions/231#discussioncomment-9522117

## Total received: 120£

**Comment by 1001ruchka** [comment]
I've seen other authors on GitHub use the buymeacoffee service.
Here's a tutorial on how to connect it to GitHub (GitHub sponsor):
https://blog.warengonzaga.com/how-to-enable-buy-me-a-coffee-to-your-open-source-project-on-github

**Comment by koebbe14** [comment]
I’m ready to donate.  How should I get payment to you?

**Reply by Purfview** [reply]
Press any donation button on the main page of the repo.
There is two buttons, one at the top and another at the bottom.

**Reply by koebbe14** [reply]
40 euros sent via PayPal.

**Reply by Purfview** [reply]
Thank you. Got the message that 40£ [not euros] received. 
If you wanted to donate other amount then you need to press "Other" button there for the custom amount.

**Comment by thbaero** [comment]
40£ done :) 
+10£ done

**Comment by 1001ruchka** [comment]
£10.00 via PayPal

**Comment by Purfview** [comment]
Someone donated 20 without any note attached, I assume it's for this cause.

Now I'll start looking how to implement this, let's hope it wouldn't take more than few days.

**Reply by Purfview** [reply]
Updated: Found an email from that donator, it wasn't for this. Anyway, I assume @dgoryeo will come through with his pledge.

**Reply by thbaero** [reply]
Thanks for the update. Let's wait a bit and see how things come along... In the meantime, thanks again for your will to implement diarization!

**Reply by dgoryeo** [reply]
I'm away on vacation. Just made my donation through a friend for 20 GBP . Thanks for making this happen.

**Reply by thbaero** [reply]
Thank you for coming through even though the goal was already met. I think you're showing your support to the importance of this idea and not just paying @Purfview 

Obviously it is just and fair to compensate Purfview for his effort for us. But according to his goal you didn't have to do it, but you did. So, thank you for that!

**Reply by dgoryeo** [reply]
Hi @Purfview , I came across this repo and I thought of sharing that with you as an idea for some future feature to consider. In this implementation the developer has added a placeholder for sample speaker voice to guide the diarization engine. I hadn't seen that approach before, so to me it was a novel approach.  Again, this is for future consideration not this round of course.

[`speaker_recognition`](https://github.com/sofiierm/speaker_recognition_whisperx_nemo)

**Comment by b1naryv1k1ng** [comment]
Hello,

Was wondering if there was any progress on this? I'd gladly throw some extra into the donation pool for this, as I'd love to be able to get the diarisation features of WhisperX

**Reply by ghost** [reply]
@b1naryv1k1ng I don't know if it's right to respond to Purfview's behalf, I have a little contact with him, I sent him a message to chat the other night when he had an illness, it's not my place to tell him about his illness, but he was still in the recovery phase, so I don't think there has been any progress.

**Reply by b1naryv1k1ng** [reply]
Oh! I was not aware. Obviously health comes first. Hope recovery goes well.

**Reply by Elviseras** [reply]
Hi,
Is this thread/development still active?
I would like to donate too, but there seems to be no progress and this development has stopped....

**Reply by Purfview** [reply]
>Is this thread/development still active?

Covid unexpectedly threw me out from my schedules, I had to put this project at the bottom of my todo list, but now I'm back at it, you can see it from my [last activity](https://github.com/Purfview/whisper-standalone-win/issues/299) on this repo, so now you can expect this implemented sooner than latter. Donation will be appreciated.

And thanks to all other donators.

**Reply by Elviseras** [reply]
Hi Purfview,
I'm glad to know that you are now recovered!! This is the most important thing... 
Then for my part I will make the donation as soon as I have a free moment.. 
Thank you for info and good news!!

**Reply by b1naryv1k1ng** [reply]
Glad you're feeling better!

**Reply by thbaero** [reply]
Couldn't agree more. So glad to hear about your recovery. Take your time and don't stress yourself. Health is much more important than a lot of other things including speaker detection ;)

**Comment by Purfview** [comment]
Implemented in v192.1
CUDA speed probably could be improved.

Post there: https://github.com/Purfview/whisper-standalone-win/discussions/322

**Reply by Elviseras** [reply]
Thank you very much! 
Could you tell us what other parameters or settings we have to customize the diarization process?

**Reply by thbaero** [reply]
You're the best! Thank you!
🙌

**Reply by whitelotusapps** [reply]
Thank you, to all involved! 🙏

**Reply by Purfview** [reply]
> Could you tell us what other parameters or settings we have to customize the diarization process?

Not much atm, --speaker, --diarize_device
Use `--help` to see all commands.
